packages/assets-controllers/src/TokensController.ts

```diff
@@ -667,144 +611,32 @@ export class TokensController extends BaseController<
-              return resolve(meta.asset.address);
-            case SuggestedAssetStatus.rejected:
-              return reject(new Error('User rejected to watch the asset.'));
-            case SuggestedAssetStatus.failed:
-              return reject(new Error(meta.error.message));
-            /* istanbul ignore next */
-            default:
-              return reject(new Error(`Unknown status: ${meta.status}`));
-          }
-        },
-      );
-    });
-
-    const { suggestedAssets } = this.state;
-    suggestedAssets.push(suggestedAssetMeta);
-    this.update({ suggestedAssets: [...suggestedAssets] });
+    validateTokenToWatch(asset);
 
-    this._requestApproval(suggestedAssetMeta);
+    await this._requestApproval(suggestedAssetMeta);
 
-    return { result, suggestedAssetMeta };
-  }
-
-  /**
-   * Accepts to watch an asset and updates it's status and deletes the suggestedAsset from state,
-   * adding the asset to corresponding asset state. In this case ERC20 tokens.
-   * A `<suggestedAssetMeta.id>:finished` hub event is fired after accepted or failure.
-   *
-   * @param suggestedAssetID - The ID of the suggestedAsset to accept.
-   */
-  async acceptWatchAsset(suggestedAssetID: string): Promise<void> {
-    const { selectedAddress } = this.config;
-    const { suggestedAssets } = this.state;
-    const index = suggestedAssets.findIndex(
-      ({ id }) => suggestedAssetID === id,
-    );
-    const suggestedAssetMeta = suggestedAssets[index];
-    try {
-      switch (suggestedAssetMeta.type) {
-        case 'ERC20':
-          const { address, symbol, decimals, image } = suggestedAssetMeta.asset;
-          await this.addToken(
-            address,
-            symbol,
-            decimals,
-            image,
-            suggestedAssetMeta?.interactingAddress || selectedAddress,
-          );
-
-          this._acceptApproval(suggestedAssetID);
-
-          const acceptedSuggestedAssetMeta = {
-            ...suggestedAssetMeta,
-            status: SuggestedAssetStatus.accepted,
-          };
-          this.hub.emit(
-            `${suggestedAssetMeta.id}:finished`,
-            acceptedSuggestedAssetMeta,
-          );
-          break;
-        default:
-          throw new Error(
-            `Asset of type ${suggestedAssetMeta.type} not supported`,
-          );
-      }
-    } catch (error) {
-      this.failSuggestedAsset(suggestedAssetMeta, error);
-
-      this._rejectApproval(suggestedAssetID);
-    }
-
-    const newSuggestedAssets = suggestedAssets.filter(
-      ({ id }) => id !== suggestedAssetID,
-    );
-    this.update({ suggestedAssets: [...newSuggestedAssets] });
-  }
-
-  /**
-   * Rejects a watchAsset request based on its ID by setting its status to "rejected"
-   * and emitting a `<suggestedAssetMeta.id>:finished` hub event.
-   *
-   * @param suggestedAssetID - The ID of the suggestedAsset to accept.
-   */
-  rejectWatchAsset(suggestedAssetID: string) {
```

Comment:
Nothing needs to happen now when rejecting since there's no longer status for `suggestedAssets`, so this is dead code.